### PR TITLE
v2.0.1 mainnet upgrade

### PR DESCRIPTION
**Pull request by lezzokafka**
**Do not merge this before the network has been upgraded.**

### CODE REVIEW

**Comment by tomtau** [comment]
for `Hardware::CPU.intel?` -- should this check be also on Mac + have else branches for arm?

File: chain-maind.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
   desc "chain-main daemon"
   homepage "https://github.com/crypto-org-chain/chain-main"
-  version "1.2.1"
+  version "2.0.1"
   bottle :unneeded
 
   if OS.mac?
-    url "https://github.com/crypto-org-chain/chain-main/releases/download/v1.2.1/chain-main_1.2.1_Darwin_x86_64.tar.gz"
-    sha256 "4004c08609598b63c14e1611c174fa55c1425374a0d923b7a4a34d4792101fb8"
+    url "https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_x86_64.tar.gz"
```
Comment (allthatjazzleo) [inline chain-maind.rb]:
```suggestion
    if Hardware::CPU.arm?
      url "https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_arm64.tar.gz"
      sha256 "eb6274724eab1957a7b708a5a0887de9f213d053841fb58041f697e13728a454"
    elsif Hardware::CPU.intel?
      url "https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_x86_64.tar.gz"
      sha256 "a4d51bf98350c7ecbb5e6bab192c9cac2f4059754e5507d2a1970a8a5488c74a"
    end
```

File: chain-maind.rb [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    sha256 "a4d51bf98350c7ecbb5e6bab192c9cac2f4059754e5507d2a1970a8a5488c74a"
```
Comment (allthatjazzleo) [inline chain-maind.rb]:
```suggestion
```

File: chain-maind.rb [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
   elsif OS.linux?
     if Hardware::CPU.intel?
```
Comment (allthatjazzleo) [inline chain-maind.rb]:
```suggestion
    if Hardware::CPU.arm?
      url "https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_arm64.tar.gz"
      sha256 "345c7eacfc768df355c3b3ecadc9cc39e3c1656c9c0c4d465b938834fa66ff03"
    else
```

File: chain-maind.rb [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-      url "https://github.com/crypto-org-chain/chain-main/releases/download/v1.2.1/chain-main_1.2.1_Linux_x86_64.tar.gz"
-      sha256 "25a5f249e0c4764cc44353d35cd783c71ec1b0e133415ed0a845a83eb012bac4"
+      url "https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_x86_64.tar.gz"
```
Comment (allthatjazzleo) [inline chain-maind.rb:12]:
```suggestion
```

File: chain-maind.rb (hunk 1 above)
Comment (tomtau) [reply chain-maind.rb]: `elsif Hardware::CPU.intel?`

**Pull request merged**